pkg/providers/v2/cloud.go

```diff
@@ -107,10 +173,191 @@ func newCloud() (cloudprovider.Interface, error) {
 		return nil, err
 	}
 
-	return &cloud{
-		creds:  creds,
-		region: region,
-	}, nil
+	instances, err := newInstances(region, creds)
+	if err != nil {
+		return nil, err
+	}
+
+	ec2Sess, err := session.NewSession(&aws.Config{
+		Region:      aws.String(region),
+		Credentials: creds,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("unable to initialize AWS session: %v", err)
+	}
+	
+	ec2Service := ec2.New(ec2Sess)
+	if err != nil {
+		return nil, fmt.Errorf("error creating AWS ec2 client: %q", err)
+	}
+
+	awsCloud := &cloud{
+		creds:     creds,
+		instances: instances,
+		region:    region,
+		cfg:       &cfg,
+		metadata:  metadataClient,
+		ec2:       ec2Service,
+	}
+
+	selfAWSInstance, err := awsCloud.buildSelfAWSInstance()
```

Comment:
"selfAWSInstance" was only required in-tree because kubelet had to reference it's own instance. In the context here I don't think we need it.